Update `Snap geometry edits` sample to include geometry guides (#234)
Esri/arcgis-maps-sdk-kotlin-samples: snap-geometry-edits/README.md

To create a geometry, press the create button to choose the geometry type you want to create (i.e. points, multipoints, polyline, or polygon) and interactively tap and drag on the map view to create the geometry.

> Snap settings can be configured by enabling and disabling snapping, feature snapping, geometry guides and snap sources.

Reviewer: FYI I believe it's standard to use the oxford comma in lists like this that end with a conjunction (i.e. use a comma before the "and"):

```suggestion
Snap settings can be configured by enabling and disabling snapping, feature snapping, geometry guides, and snap sources.
```

4. Create a `ReticleVertexTool` and set it into the `GeometryEditor.tool`.
5. Call `syncSourceSettings` after the map's operational layers are loaded and the geometry editor connected to the map view.
6. Set `SnapSettings.isEnabled` and `SnapSourceSettings.isEnabled` to true for the `SnapSource` of interest.

> 7. Toggle geometry guides using `SnapSettings.isGeometryGuidesEnabled` and feature snapping using `SnapSettings.isFeatureSnappingEnabled`.

Reviewer: I'd suggest changing "toggle" to something more definitive here, e.g. 
```suggestion
7. Enable or disable geometry guides using `SnapSettings.isGeometryGuidesEnabled` and feature snapping using `SnapSettings.isFeatureSnappingEnabled`.
```

Snapping can be used during interactive edits that move existing vertices using the `VertexTool` or `ReticleVertexTool`. It is also supported for adding new vertices for input devices with a hover event (such as a mouse move without a mouse button press). Using the `ReticleVertexTool` to add and move vertices allows users of touch screen devices to clearly see the visual cues for snapping.

> Geometry guides are enabled by default when snapping is enabled. These allow for snapping to a point coinciding with, parallel to, perpendicular to or extending an existing geometry.

Reviewer: ```suggestion
Geometry guides are enabled by default when snapping is enabled. These allow for snapping to a point coinciding with, parallel to, perpendicular to, or extending an existing geometry.
```
